src/Security/Authenticator.php

```diff
@@ -117,9 +117,10 @@ public static function register_authenticator($authenticator)
             return false;
         }
 
-        if (in_array($authenticator, self::$authenticators) == false) {
-            if (call_user_func(array($authenticator, 'on_register')) === true) {
-                array_push(self::$authenticators, $authenticator);
+        $authenticators = self::config()->get('authenticators');
+        if (in_array($authenticator, $authenticators) == false) {
+            if ($authenticator::on_register() === true) {
+                self::config()->merge('authenticators', [$authenticator]);
```

Comment:
Yep, I added this more as a fallback so it wouldn't immediately break any existing `register()` calls from other code, but the idea is to deprecate it and move to `Config` only. I'm in that regard slightly concerned about the `unregister` method though.